actor/src/main/scala/org/apache/pekko/pattern/AskSupport.scala

```diff
@@ -617,7 +617,7 @@ private[pekko] final class PromiseActorRef(
 
   override def !(message: Any)(implicit sender: ActorRef = Actor.noSender): Unit = state match {
     case Stopped | _: StoppedWithPath =>
-      provider.deadLetters ! message
+      provider.deadLetters ! DeadLetter(message, if (sender eq Actor.noSender) provider.deadLetters else sender, this)
```

Comment:
In fact, the sender will always be empty because in typed, the passing of the sender is changed from explicitly on method calling to optional choice as a field  in the message, unless we can extract the value with the same field name (such as sender, replyTo) from the user's message and use it here.